### ISSUE TITLE
v2.3.3 breaks NAT on Ubuntu 22.04 + Vbox 6.1

### ISSUE THREAD

**Issue by matlink** [hashicorp/vagrant#13015]
Since upgrading vagrant from 2.3.2 to 2.3.3, vagrant up fails complaining it can't forward port on the first NAT interface : 
```
# vagrant init hashicorp/bionic64
A `Vagrantfile` has been placed in this directory. You are now
ready to `vagrant up` your first virtual environment! Please read
the comments in the Vagrantfile as well as documentation on
`vagrantup.com` for more information on using Vagrant.
# vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'hashicorp/bionic64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'hashicorp/bionic64' version '1.0.282' is up to date...
==> default: Setting the name of the VM: tmp_default_1670181016141_7987
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
The adapter to attach a forwarded port to was not found. Please
verify that the given adapter is setup on the machine as a NAT
interface.

Host port: 2222
Guest port: 22
Adapter: 1
```
Downgrading to 2.3.2 fixes the issue.
### Debug output
https://gist.github.com/matlink/8ca8fa35ed5860df9061843bbfbcb3e4

**Comment by chrisroberts** [comment]
Hi there,

I'm attempting to reproduce the error you are encountering but am not seeing the same behavior. Can you please provide a minimal Vagrantfile that reproduces this error? It would be very helpful in tracking down the underlying cause.

Thanks!

**Comment by matlink** [comment]
The one generated with init is enough for me to reproduce:
```
Vagrant.configure("2") do |config|
  config.vm.box = "hashicorp/bionic64"
end
```

**Comment by jeon3029** [comment]
I got the same response but I cleared by this method.

1. clear the virtual box and reinstall to [VirtualBox 7.0.4]

2. clear the box image 

rm -rf ~/"VirtualBox VMs"

4. clear the vagrant default settings

rm -rf ./.vagrant

5. replay

vagrant init
vagrant up

----

this is my default nat settings

<img width="666" alt="image" src="https://user-images.githubusercontent.com/10026516/218352143-ef2eb569-ad90-46cb-83ae-9d53c176a592.png">
